### ISSUE TITLE
Cannot write to stdin after command execution

### ISSUE THREAD

**Issue by adam3039** [sshnet/SSH.NET#439]
I'm trying to run a program that requires an input to select which module within the program to run. I can connect fine, and I can run the program, but I seem to be having zero luck writing to the shellstream I am creating. I've gone through every issue ticket regarding this, and most of the solutions make it seem super simple, but nothing is working for me. 

```
using (var client = new SshClient("10.0.0.136", "ubuntu", "ubuntu"))
{
       client.Connect();
       var runCommand = client.CreateCommand("cd Documents/OnboardSDK/build/bin; ./sample-program");

       var result = runCommand.BeginExecute();
       var reader = new StreamReader(runCommand.OutputStream);
       Console.WriteLine(reader.ReadToEnd());
                
       var stream = client.CreateShellStream("term", 80, 24, 800, 600, 1024);
       stream.WriteLine("a");

       client.Disconnect();
}
```

I get the output of the program up until where the stdin should be, but writing to the stream isn't causing the program to continue. "a" is the option I am picking, and in a putty session all I need to do is type a and hit the enter key for the program to continue.

Any insight is greatly appreciated, I am new to this library. Thanks.

**Comment by DanielVukelich** [comment]
Are you running a curses-type TUI program by chance?  Those don't read from stdin the same way that CLI programs do.

Try running this command and sending text to it:
 `echo $(</dev/stdin) > capturedInput.txt`
It will log what you send via stdin to capturedInput.txt which could help verify where the issue is

**Comment by adam3039** [comment]
Hi @DanielVukelich 

Thanks for the idea. I don't believe it's a curses-type TUI program. Although I'm wondering if possibly it has anything to do with it being an std::cin (not stdin)

This is the bit of source code from the program I am trying to use, where it takes in a character option:

```
 // Display interactive prompt
  std::cout
    << "| Available commands:                                            |"
    << std::endl;
  std::cout
    << "| [a] Monitored Takeoff + Landing                                |"
    << std::endl;
  std::cout
    << "| [b] Monitored Takeoff + Position Control + Landing             |"
    << std::endl;
  char inputChar;
  std::cin >> inputChar;
```

Admittedly I'm not a C++ or Linux wizard, so I may be missing something obvious. Any insight is really appreciated. My alternative option is just modifying the script so that it doesn't require inputs, but I'd like to get this figured out just for a better general understanding.

**Comment by IgorMilavec** [comment]
@adam3039 I'm trying to categorize this issue, but it is not clear to me what the root cause for this problem could be. Were you able to determine what is causing this? Can you create a repro (using some standard shell program)?

**Comment by WojciechNagorski** [comment]
Fixed by #1293
